Kudu.Core/Jobs/BaseJobRunner.cs

```diff
@@ -102,10 +89,14 @@ private void CacheJobBinaries(JobBase job, IJobLogger logger)
             {
                 try
                 {
-                    int currentHash = CalculateHashForJob(JobBinariesPath);
-                    int lastHash = CalculateHashForJob(WorkingDirectory);
-
-                    if (lastHash == currentHash)
+                    // If the newest file in the WebJobs folder is newer than the newest file in the cache,
+                    // treat the cache as invalid.
+                    // Note that we don't want to invalidate the cache if files are added in the cache. Also, we
+                    // update the .exe.config file, and we don't want that to invalidate the cache.
+                    DateTime lastWebJobsWriteTime = FileSystemHelpers.GetFolderLatestWriteTimeUtc(JobBinariesPath);
+                    DateTime lastWorkingDirectoryWriteTime = FileSystemHelpers.GetFolderLatestWriteTimeUtc(WorkingDirectory);
+
+                    if (lastWebJobsWriteTime <= lastWorkingDirectoryWriteTime)
```

Comment:
Out of curiosity, when the source's time is newer than destination's time, do we do a smart copy (only copy those newer, remove those being deleted etc)?  Given it is more expensive to read the content than to read attribute (last modified time etc).